drivers/pinmux/pinmux_mchp_xec.c

```diff
@@ -123,11 +128,82 @@ static const struct pinmux_driver_api pinmux_xec_driver_api = {
 };
 
 #define PINMUX_ADDR(n) DT_REG_ADDR(DT_PHANDLE(DT_NODELABEL(n), ph_reg))
+#define PINMUX_PORT_NUM(n) DT_PROP(DT_NODELABEL(n), port_num)
+
+/*
+ * Chip DTSI has
+ * / {
+ * ...
+ * pinmux {
+ * 	compatible = "microchip,xec-pinmux";
+ *
+ *	pinmux_000_036: pinmux-0 {
+ *		ph-reg = <&gpio_000_036>;
+ *		port-num = <0>;
+ *	};
+ * ...
+ *	pinmux_240_276: pinmux-5 {
+ *		ph-reg = <&gpio_240_276>;
+ *		port-num = <5>;
+ *	};
+ * };
+ * ...
+ * soc {
+ * ...
+ * };
+
+ * How do we use DT_FOREACH_xxxx when pinmux is not under soc, is not in
+ * the form instance_name: class_name@address { properties }; ?
+ */
```

Comment:
You can use DT_INST to get the pinux list, since you have the compatible string "microchip,xec-pinmux", and then DT_FOREAHC_CHILD... at that point to iterate thru the children

Reply:
@jackrosenthal Found the issue.  DT macro failures caused by chip DTSI syntax:
pinmux { ... };
DT macros require by guess work looking at many other SoC's
pinmux: pinmux { ... };